dataset/batch_image.py

```diff
@@ -1,525 +1,767 @@
-                new_image = self._preserve_shape(new_image, image_size, crop)
-        else:
-            new_image = image
-        return new_image
+        if fmt == 'image':
+            self._dump_image(dst, components)
+        return super().dump(dst, fmt, components, *args, **kwargs)
 
-    def _preserve_shape(self, image, shape, crop=CROP_CENTER):
-        """ Change the image shape by cropping and adding empty pixels to fit the given shape """
-        raise NotImplementedError()
 
-    @action
-    @inbatch_parallel(init='indices', post='assemble')
-    def rotate(self, ix, components='images', angle=0, preserve_shape=True, **kwargs):
-        """ Rotate all images in the batch at the given angle
+@transform_actions(prefix='_', suffix='_all', wrapper='apply_transform_all')
+@transform_actions(prefix='_', suffix='_', wrapper='apply_transform')
+class ImagesBatch(BaseImagesBatch):
+    """ Batch class for 2D images.
 
-        Parameters
-        -----------
-        components : str
-            a component name or names which data should be rotated
+    Images are stored as numpy arrays (N, H, W, C).
+    """
 
-        angle : float
-            the rotation angle in degrees.
-        preserve_shape : bool
-            whether to keep shape after rotating
-            (always True for images as arrays, can be False for PIL.Images)
-        """
-        return self._rotate_one(ix, components, angle, preserve_shape, **kwargs)
+    @classmethod
+    def _get_image_shape(cls, image):
+        return image.shape[:2]
+
+    @property
+    def image_shape(self):
+        """: tuple - shape of the image """
+        if isinstance(self.images.dtype, object):
+            _, shapes_count = np.unique([image.shape for image in self.images], return_counts=True, axis=0)
+            if len(shapes_count) == 1:
+                return self.images.shape[1:]
+            else:
+                raise RuntimeError('Images have different shapes')
+        return self.images.shape[1:]
 
-    @action
     @inbatch_parallel(init='indices', post='assemble')
-    def random_rotate(self, ix, components='images', p=1., angle=None, **kwargs):
-        """ Rotate each image in the batch at a random angle
+    def _load_image(self, ix, src=None, components="images"):
+        """ Wrapper for scipy.ndimage.open.
+
+        .. note:: only works with a single component
 
         Parameters
-        -----------
+        ----------
+        path : str, None
+        ix : str
+            element's index (filename)
         components : str
-            a component name or names which data should be rotated
+            component to load
 
-        p : float
-            a probability to apply rotate
-            (0. - don't rotate, .5 - rotate half of images, 1 - rotate all images)
-
-        angle : tuple
-            an angle range in the form of (min_angle, max_angle), in degrees
+        Returns
+        -------
+        np.ndarray : Loaded image
```

Comment:
This docstring is for the decorated _load_image method which returns a batch

Reply:
+